source/mesh_deformation/interface.cc

```diff
@@ -464,6 +483,121 @@ namespace aspect
 
 
 
+    template <int dim>
+    AffineConstraints<double> MeshDeformationHandler<dim>::make_initial_constraints()
+    {
+      AssertThrow(this->get_parameters().mesh_deformation_enabled, ExcInternalError());
+
+      // initial_deformation_constraints can use the same hanging node
+      // information that was used for mesh_vertex constraints.
+      AffineConstraints<double> initial_deformation_constraints(mesh_locally_relevant);
+      initial_deformation_constraints.merge(mesh_vertex_constraints);
+
+      // Add the vanilla periodic boundary constraints
+      std::set< types::boundary_id > periodic_boundaries;
+      using periodic_boundary_pairs = std::set< std::pair< std::pair<types::boundary_id, types::boundary_id>, unsigned int> >;
+      periodic_boundary_pairs pbp = this->get_geometry_model().get_periodic_boundary_pairs();
+      for (periodic_boundary_pairs::iterator p = pbp.begin(); p != pbp.end(); ++p)
```

Comment:
Maybe just use ``for (const auto& p : this->get_geometry_model().get_periodic_boundary_pairs())``?

Reply:
I specifically chose not to in this case, because the type is so complicated that an `auto` is very confusing (it is near impossible to tell what the type is without going digging into the return type of the geometry model function). This is an aspect of the `auto` keyword that I think is not optimal.